src/scenarios/shared/runner.py

```diff
@@ -328,48 +338,129 @@ def run(self):
                 const.TRACEDIR,
                 '-v'
             ]
+            RunCommand(installCmd, verbose=True).run()
 
-            RunCommand(cmdline, verbose=True).run()
-
+            getLogger().info("Completed install, running shell.")
+            cmdline = [ 
+                adb.stdout.strip(),
+                'shell',
+                f'cmd package resolve-activity --brief {self.packagename} | tail -n 1'
+            ]
+            getActivity = RunCommand(cmdline, verbose=True)
+            getActivity.run()
+            getLogger().info(f"Target Activity {getActivity.stdout}")
+
+            # More setup stuff
+            checkScreenOnCmd = [ 
+                adb.stdout.strip(),
+                'shell',
+                f'dumpsys input_method | grep mInteractive'
+            ]
+            checkScreenOn = RunCommand(checkScreenOnCmd, verbose=True)
+            checkScreenOn.run()
+
+            keyInputCmd = [
+                adb.stdout.strip(),
+                'shell',
+                'input',
+                'keyevent'
+            ]
 
+            if("mInteractive=false" in checkScreenOn.stdout): 
+                # Turn on the screen to make interactive and see if it worked
+                getLogger().info("Screen was off, turning on.")
+                screenWasOff = True
+                RunCommand(keyInputCmd + ['26'], verbose=True).run() # Press the power key
+                RunCommand(keyInputCmd + ['82'], verbose=True).run() # Unlock the screen with menu key (only works if it is not a password lock)
+
+                checkScreenOn = RunCommand(checkScreenOnCmd, verbose=True)
+                checkScreenOn.run()
+                if("mInteractive=false" in checkScreenOn.stdout):
+                    getLogger().exception("Failed to make screen interactive.")
+
+            # Actual testing some run stuff
+            getLogger().info("Test run to check if permissions are needed")
+            activityname = getActivity.stdout
+
+            startAppCmd = [ 
+                adb.stdout.strip(),
+                'shell',
+                'am',
+                'start-activity',
+                '-W',
+                '-n',
+                activityname
+            ]
+            testRun = RunCommand(startAppCmd, verbose=True)
+            testRun.run()
+            testRunStats = re.findall(runSplitRegex, testRun.stdout) 
+            getLogger().info(f"Test run activity: {testRunStats[3]}")
```

Comment:
What is testRunStats[3]? Can you break down what the different groups of this regex are.

Reply:
Added a break down the data split at the top of this switch section (line 310)